src/otx/algo/instance_segmentation/otx_instseg_evaluation.py

```diff
@@ -0,0 +1,67 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Customised MAP metric for instance segmentation."""
+
+from __future__ import annotations
+
+from typing import Any
+
+import pycocotools.mask as mask_utils
+import torch
+from torchmetrics.detection.mean_ap import MeanAveragePrecision
+
+
+class OTXInstSegMeanAveragePrecision(MeanAveragePrecision):
```

Comment:
Would you please add a unit test which can verify that this metric class can produce the same outputs as `torchmetrics.detection.mean_ap.MeanAveragePrecision`

Reply:
Added in `tests/unit/algo/instance_segmentation/test_evaluation.py`